ray-operator/test/e2e/support.go

```diff
@@ -175,3 +179,50 @@ func jobSubmitterPodTemplateApplyConfiguration() *corev1ac.PodTemplateSpecApplyC
 						corev1.ResourceMemory: resource.MustParse("500Mi"),
 					}))))
 }
+
+func redisDeploymentApplyConfigurationWithoutPassword(namespace string) *appsv1ac.DeploymentApplyConfiguration {
+	return redisDeploymentApplyConfiguration(namespace, "")
+}
+
+func redisDeploymentApplyConfiguration(namespace string, password string) *appsv1ac.DeploymentApplyConfiguration {
+	redisContainer := corev1ac.Container().WithName("redis").WithImage("redis:7.4").
+		WithPorts(corev1ac.ContainerPort().WithContainerPort(6379))
+	if password != "" {
+		redisContainer.WithCommand("redis-server", "--requirepass", password)
+	}
+
+	return appsv1ac.Deployment("redis", namespace).
+		WithSpec(appsv1ac.DeploymentSpec().
+			WithReplicas(1).
+			WithSelector(metav1ac.LabelSelector().WithMatchLabels(map[string]string{"app": "redis"})).
+			WithTemplate(corev1ac.PodTemplateSpec().
+				WithLabels(map[string]string{"app": "redis"}).
+				WithSpec(corev1ac.PodSpec().
+					WithContainers(redisContainer),
+				),
+			),
+		)
+}
+
+func redisServiceApplyConfiguration(namespace string) *corev1ac.ServiceApplyConfiguration {
+	return corev1ac.Service("redis", namespace).
+		WithSpec(corev1ac.ServiceSpec().
+			WithSelector(map[string]string{"app": "redis"}).
+			WithPorts(corev1ac.ServicePort().
+				WithPort(6379),
+			),
+		)
+}
+
+func checkEnv(test Test, namespace string, env string, rayClusterAC *rayv1ac.RayClusterApplyConfiguration) func(g Gomega) bool {
+	return func(g Gomega) bool {
+		rayCluster, err := test.Client().Ray().RayV1().RayClusters(namespace).Apply(test.Ctx(), rayClusterAC, TestApplyOptions)
```

Comment:
why do we need to apply the apply config again after the cluster is created?

Reply:
That was copied from my previous commit: https://github.com/ray-project/kuberay/blob/9068102246eeb5ab9d9e0b9a7480618d3f348686/ray-operator/test/e2e/raycluster_test.go#L181

Indeed we don't need to apply it again. Could you help change it to
```go
g.Eventually(HeadPod(test, rayCluster), TestTimeoutShort).Should(WithTransform(IsEnvVarExisted(utils.RAY_REDIS_ADDRESS), BeTrue()))
```
```go
func IsEnvVarExisted(env string) func(pod *corev1.Pod) bool {
	return func(pod *corev1.Pod) bool {
		for _, envVar := range pod.Spec.Containers[utils.RayContainerIndex].Env {
			if envVar.Name == env {
				return true
			}
		}
		return false
	}
}

```

Reply:
Fixed along with the next feedback. https://github.com/ray-project/kuberay/pull/2731/commits/3e938ed36ea4abf839e81d8b313944b7727b8067 https://github.com/ray-project/kuberay/pull/2731/commits/c3747a68d2ce8061237b615f0a2b7ce793aee0b3